crates/ruff_cache/src/cache_key.rs

```diff
@@ -391,3 +363,88 @@ impl CacheKey for Pattern {
         self.as_str().cache_key(state);
     }
 }
+
+#[derive(Clone, Default)]
+pub struct CacheKeyHasher {
+    inner: SeaHasher,
+}
+
+impl CacheKeyHasher {
+    pub fn new() -> Self {
+        Self {
+            inner: SeaHasher::new(),
+        }
+    }
+}
+
+impl Hasher for CacheKeyHasher {
```

Comment:
I replaced the `Deref` implementation with implementing `Hasher` directly to avoid changing the hasher to be a breaking API change.